core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/AST/AndPropertyShape.java

```diff
@@ -288,9 +290,22 @@ public String buildSparqlValidNodes(String targetVar) {
 
 	@Override
 	public Stream<StatementPattern> getStatementPatterns() {
-		return and
-				.stream()
+		StatementPattern subject = new StatementPattern(
+				new Var("?this"),
```

Comment:
Typically variable names do not actually start with a question mark - in SPARQL they do of course but our query parsers all treat it like a marker prefix, not as part of the actual variable name.

Reply:
This PR is mostly to fix a bug in the old AST so that I can handel it properly in the new AST. This is essentially just copy-paste from other parts of the code that are built like this, so I don't want to start fixing it up now. I will take a look in the new AST and see if I prefix my vars with `?` or not. And also check the random UUID thing you commented.